File: Collatz.java
// CS1010 AY2016/7 Semester 1
// PE1 Ex1: collatz.c

import java.util.*;

public class Collatz { // Class names are in UpperCamelCase

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter positive numbers for n and count: ");  
           
        int n = sc.nextInt();
        int count = sc.nextInt();
        
        System.out.println(); // equivalent to System.out.print("\n");
        
        // display the internal execution of Collatz's sequence
        displayCollatzInternal(n, count);
    }
    
    public static void displayCollatzInternal(int n, int count) {
        System.out.print("The collatz sequence is: ");
    
        // Variables in lowerCamelCase
        int divExecutedCount = 0;
        int mulExecutedCount = 0;
        
        // In Java, you can no longer use non-zero numbers to denote 'true'
        while (n != 1 && count > 0) { 
            if (n % 2 != 0) { // odd
                n = 3 * n + 1;
                mulExecutedCount++;
            } else { // even
                n /= 2;
                divExecutedCount++;
            }
            
            count--;
            System.out.print(n + " "); 
        }
        System.out.println();
        
        System.out.println("The division statement has been executed " + divExecutedCount + " times.");
        System.out.println("The multiplication statement has been executed "+ mulExecutedCount + " times.");

    }
}

File: samples/C/collatz/collatz.c
// CS1010 AY2016/7 Semester 1
// PE1 Ex1: collatz.c

#include <stdio.h>

void displayCollatzInternal(int, int);

int main(void) {
    // Get user input
    printf("Enter positive numbers for n and count: ");

    int n, count;
    scanf("%d%d", &n, &count);

    printf("\n");

    // display the internal execution of Collatz's sequence
    displayCollatzInternal(n, count);

    return 0;
}

void displayCollatzInternal(int n, int count) {    
    printf("The collatz sequence is: ");

    int div_executed_count = 0;
    int mul_executed_count = 0;

    while (n != 1 && count) {
        if (n % 2) { // odd
            n = 3 * n + 1;
            mul_executed_count++;
        } else { // even
            n /= 2;
            div_executed_count++;
        }

        count--;
        printf("%d ", n);    
    }
    printf("\n");

    printf("The division statement has been executed %d times.\n", div_executed_count);
    printf("The multiplication statement has been executed %d times.\n", mul_executed_count);
}
